source/val/Function.cpp

```diff
@@ -93,15 +99,36 @@ spv_result_t Function::RegisterLoopMerge(uint32_t merge_id,
                                          uint32_t continue_id) {
   RegisterBlock(merge_id, false);
   RegisterBlock(continue_id, false);
-  cfg_constructs_.emplace_back(get_current_block(), &blocks_.at(merge_id),
-                               &blocks_.at(continue_id));
+  BasicBlock& merge_block = blocks_.at(merge_id);
+  BasicBlock& continue_block = blocks_.at(continue_id);
+  assert(merge_block.is_type(kBlockTypeUndefined));
```

Comment:
This is always going to be true since the Undefined type bit is never reset.  ?!